helpers.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import os
+import subprocess
+from typing import Union
+
+import discord
+import requests
+
+# logs data to the discord.log file, if this file doesn't exist at runtime it is created automatically
+logger = logging.getLogger('discord')
+logger.setLevel(logging.INFO)  # logging levels: NOTSET (all), DEBUG (bot interactions), INFO (bot connected etc)
+handler = logging.FileHandler(filename='discord.log', encoding='utf-8', mode='w')
+handler.setFormatter(logging.Formatter('%(asctime)s:%(levelname)s:%(name)s: %(message)s'))
+logger.addHandler(handler)
+
+FFMPEG_LOCATION = 'ffmpeg'
+
+
+def get_reddit_json_payload(url: str) -> Union[tuple[bool, str], tuple[bool, dict]]:
+    try:
+        r = requests.get(
+            f'{url.split("?", 1)[0]}.json',
+            headers={'User-agent': 'redditBot v0.1'}
+        )
+    except requests.exceptions.RequestException:
+        return False, ':warning: Error getting request from url, it may be invalid'
+
+    if r.status_code == 429:
+        return False, ':warning: Received error `429` from Reddit'
+    else:
+        json_payload = r.json()
+        return True, {
+            'title': json_payload[0]["data"]["children"][0]["data"]["title"],
+            'subreddit': json_payload[0]["data"]["children"][0]["data"]["subreddit_name_prefixed"],
+            'video': json_payload[0]["data"]["children"][0]["data"]["secure_media"]["reddit_video"]["fallback_url"],
+            'audio': f'{json_payload[0]["data"]["children"][0]["data"]["url_overridden_by_dest"]}/DASH_audio.mp4',
+            'image': json_payload[0]["data"]["children"][0]["data"]["url_overridden_by_dest"]
+        }
+
+
+def process_media(video, audio) -> bool:
+    if requests.get(audio, headers={'User-agent': 'redditBot v0.1'}).status_code != 403:
+        process = subprocess.Popen(
+            [FFMPEG_LOCATION, '-i', f'{video}', '-i', f'{audio}', '-c', 'copy', 'output.mp4', '-y'],
+            cwd=os.getcwd(),
+            shell=True,
```

Comment:
This is running on Linux so not sure if this will work....

Reply:
I haven't tested it on my linux devenv as of yet ~~mainly because I forgot~~, gimme a mo to try it on my Debian devenv :)

Reply:
Great cheers

Reply:
righty
got a little distracted with food, removed `shell=True` that was causing issues on non-Windows environments
gonna continue eating mah spud
~~I think~~ it works fine now on Linux on my fork

Reply:
nice, thought that it might cause errors